GitUp/Application/AppDelegate.m

```diff
@@ -425,18 +424,19 @@ - (BOOL)applicationShouldOpenUntitledFile:(NSApplication*)sender {
 
 - (BOOL)applicationShouldHandleReopen:(NSApplication*)theApplication hasVisibleWindows:(BOOL)hasVisibleWindows {
   if (!hasVisibleWindows) {
-    _allowWelcome = 1;  // Always show welcome when clicking on dock icon
+    // Always show welcome when clicking on dock icon
+    [self.welcomeWindowController.model setShouldShow];
     [self handleDocumentCountChanged];
   }
   return YES;
 }
 
 - (void)applicationDidBecomeActive:(NSNotification*)notification {
-  if (_allowWelcome < 0) {
-    _allowWelcome = 1;
-  }
+//  if (_allowWelcome < 0) {
+//    _allowWelcome = 1;
+//  }
+  [self.welcomeWindowController.model setShouldShow];
```

Comment:
I think we should leave the previous mechanism in place for showing and dismissing the window.